src/mpid/ch4/shm/xpmem/xpmem_pre.h

```diff
@@ -8,9 +8,25 @@
 #define XPMEM_PRE_H_INCLUDED
 
 #include <xpmem.h>
+#include "shm_types.h"
 
 #define MPIDI_XPMEM_PERMIT_VALUE ((void *)0600)
 #define MPIDI_XPMEM_SEG_PREALLOC 8      /* Number of segments to preallocate in the "direct" block */
+#define MPIDI_XPMEM_CNT_PREALLOC 64     /* Number of shm counter to preallocate in the "direct" block */
+
+enum {
+    MPIDI_XPMEM_COPY_ALL,       /* indicate process copies all chunks in XPMEM cooperative copy */
+    MPIDI_XPMEM_COPY_ZERO,      /* indicate process copies zero chunk in XPMEM cooperative copy */
+    MPIDI_XPMEM_COPY_MIX        /* indicate both processes copies a part of chunks in XPMEM cooperative copy */
+};
+
+enum {
+    /* More explanation can be found in MPIDI_XPMEM_do_lmt_coop_copy, MPIDI_XPMEM_handle_lmt_coop_recv and
+     * MPIDI_XPMEM_ctrl_send_lmt_cts_cb routines of xpmem_recv.h and xpmem_control.c files */
+    MPIDI_XPMEM_LOCAL_FIN,      /* process sees cur_chunk == total_chunk, and local copy is done but the other
+                                 * process may be still working on copy */
+    MPIDI_XPMEM_BOTH_FIN        /* process sees cur_chunk == total_chunk + 1, cooperative copy is complete */
+};
 
```

Comment:
I'd suggest we compress the comments to something like below. What do you think?
```
/* Variables used to indicate coop copy completion cases.
 *  See more explanation in xpmem_recv.h and xpmem_control.c */
enum {
    MPIDI_XPMEM_COPY_ALL,       /* local process copied all chunks */
    MPIDI_XPMEM_COPY_ZERO,      /* local process copied zero chunk */
    MPIDI_XPMEM_COPY_MIX        /* both sides copied a part of chunks */
};

enum {
    MPIDI_XPMEM_LOCAL_FIN,      /* local copy is done but the other side may be still copying */
    MPIDI_XPMEM_BOTH_FIN        /* both sides finished copy */
};
```